amp-camp/src/server/server.js

```diff
@@ -312,4 +331,22 @@ function enableCors(req, res) {
     res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
     res.header("Access-Control-Expose-Headers", "AMP-Access-Control-Allow-Source-Origin,AMP-Redirect-To");
     res.header("AMP-Access-Control-Allow-Source-Origin", sourceOrigin);
+}
+
+// Uses mustache to render the dynamic page. Injects canonical tag, based on the requested URL.
+function renderPage(req, res, template, responseJsonObj) {
+    if(!responseJsonObj) {
+        responseJsonObj = {}
+    }
+    
+    let fullUrl = req.protocol + '://' + req.get('host') + req.originalUrl;
+    responseJsonObj.CanonicalLink = fullUrl;
+
+    mustache.tags = ['<%', '%>'];
```

Comment:
We can't keep putting this `mustache.tags = ['<%', '%>'];` line in this file again and again. This needs to be defined globally.

Reply:
In this case, the rest of the declarations have been removed, and there's a single call to mustache.tags in the new "renderPage" function, which is the only one rendering pages.
Do you think that this is not enough to have it on a single place?

Reply:
Sorry, I didn't realize that's what you'd done. My fault!